src/content-distribution/distribute/index.js

```diff
@@ -0,0 +1,209 @@
+/* globals newspack_network_distribute */
+
+/**
+ * WordPress dependencies.
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { sprintf, __, _n } from '@wordpress/i18n';
+import { useState, useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { PluginSidebar } from '@wordpress/editor';
+import { Panel, PanelBody, CheckboxControl, TextControl, Button } from '@wordpress/components';
+import { globe } from '@wordpress/icons';
+import { registerPlugin } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies.
+ */
+import './style.scss';
+
+const networkSites = newspack_network_distribute.network_sites;
+const distributedMetaKey = newspack_network_distribute.distributed_meta;
+
+function Distribute() {
+	const [ search, setSearch ] = useState( '' );
+	const [ isDistributing, setIsDistributing ] = useState( false );
+	const [ distribution, setDistribution ] = useState( [] );
+	const [ siteSelection, setSiteSelection ] = useState( [] );
+
+	const { postId, savedUrls, hasChangedContent, isSavingPost, isCleanNewPost } = useSelect( select => {
+		const {
+			getCurrentPostId,
+			getCurrentPostAttribute,
+			hasChangedContent,
+			isSavingPost,
+			isCleanNewPost,
+		} = select( 'core/editor' );
+		return {
+			postId: getCurrentPostId(),
+			savedUrls: getCurrentPostAttribute( 'meta' )?.[ distributedMetaKey ] || [],
+			hasChangedContent: hasChangedContent(),
+			isSavingPost: isSavingPost(),
+			isCleanNewPost: isCleanNewPost(),
+		};
+	} );
+
+	useEffect( () => {
+		setSiteSelection( [] );
+	}, [ postId ] );
+
+	useEffect( () => {
+		setDistribution( savedUrls );
+	}, [ savedUrls ] );
+
+	const { savePost } = useDispatch( 'core/editor' );
+	const { createNotice } = useDispatch( 'core/notices' );
+
+	const sites = networkSites.filter( url => url.includes( search ) );
+
+	const selectableSites = networkSites.filter( url => ! distribution.includes( url ) );
+
+	const getFormattedSite = site => {
+		const url = new URL( site );
+		return url.hostname;
+	}
+
+	const distribute = () => {
+		setIsDistributing( true );
+		apiFetch( {
+			path: `newspack-network/v1/content-distribution/distribute/${ postId }`,
+			method: 'POST',
+			data: {
+				urls: siteSelection,
+			},
+		} ).then( urls => {
+			setDistribution( urls );
+			setSiteSelection( [] );
+			createNotice(
+				'info',
+				sprintf(
+					_n(
+						'Post distributed with one connection.',
```

Comment:
Small language nit: Should this be "Post distributed to"? Distribute with sounds more like shared with

Reply:
You're right, it should be "to" cc @miguelpeixe

Reply:
Updated in 8b9bccc6e720c68da37b374bd98318c500b7db41